airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/util/MySqlSslConnectionUtils.java

```diff
@@ -0,0 +1,163 @@
+  public static final String PARAM_CLIENT_KEY = "client_key";
+  public static final String TRUST_KEY_STORE_URL = "trustCertificateKeyStoreUrl";
+  public static final String TRUST_KEY_STORE_PASS = "trustCertificateKeyStorePassword";
+  public static final String CLIENT_KEY_STORE_URL = "clientCertificateKeyStoreUrl";
+  public static final String CLIENT_KEY_STORE_PASS = "clientCertificateKeyStorePassword";
+  public static final String CUSTOM_TRUST_STORE = "customtruststore.jks";
+  public static final String CUSTOM_KEY_STORE = "customkeystore.jks";
+  public static final String SSL_MODE = "sslMode";
+  public static final String DISABLE = "disable";
+  public static final String VERIFY_CA = "VERIFY_CA";
+  public static final String VERIFY_IDENTITY = "VERIFY_IDENTITY";
+  public static final String ROOT_CERTIFICARE_NAME = "ca-cert.pem";
+  public static final String ROOT_CERTIFICARE_DER_NAME = "ca-cert.der";
+  public static final String CLIENT_CERTIFICARE_NAME = "client-cert.pem";
+  public static final String CLIENT_KEY_NAME = "client-key.pem";
+  public static final String CLIENT_CERT_P12 = "certificate.p12";
+
+  public static Map<String, String> obtainConnectionOptions(final JsonNode encryption) {
+    Map<String, String> additionalParameters = new HashMap<>();
+    if (!encryption.isNull()) {
+      final var method = encryption.get(PARAM_MODE).asText().toUpperCase();
+      String sslPassword = encryption.has(PARAM_CLIENT_KEY_PASSWORD) ? encryption.get(PARAM_CLIENT_KEY_PASSWORD).asText() : "";
+      var keyStorePassword = RandomStringUtils.randomAlphanumeric(10);
+      if (!sslPassword.isEmpty()) {
+        keyStorePassword = sslPassword;
+      }
+      switch (method) {
+        case VERIFY_CA -> {
+          additionalParameters.putAll(obtainConnectionCaOptions(encryption, method, keyStorePassword));
+        }
+        case VERIFY_IDENTITY -> {
+          additionalParameters.putAll(obtainConnectionFullOptions(encryption, method, keyStorePassword));
+        }
+      }
+    }
+    return additionalParameters;
+  }
+
+  private static Map<String, String> obtainConnectionFullOptions(final JsonNode encryption,
+                                                                 final String method,
+                                                                 final String clientKeyPassword) {
+    Map<String, String> additionalParameters = new HashMap<>();
+    try {
+      convertAndImportFullCertificate(encryption.get(PARAM_CA_CERTIFICATE).asText(),
+          encryption.get(PARAM_CLIENT_CERTIFICATE).asText(),
+          encryption.get(PARAM_CLIENT_KEY).asText(), clientKeyPassword);
+    } catch (final IOException | InterruptedException e) {
+      throw new RuntimeException("Failed to import certificate into Java Keystore");
+    }
+    additionalParameters.put(TRUST_KEY_STORE_URL, "file:" + CUSTOM_TRUST_STORE);
+    additionalParameters.put(TRUST_KEY_STORE_PASS, clientKeyPassword);
```

Comment:
we shouldn't reuse the password (trust store can use a random password as it won't typically hold any private information)

Reply:
ok, I changed it to random password generation to avoid duplicate password for two keystores.